src/runtime/io.cpp

```diff
@@ -340,6 +337,8 @@ extern "C" LEAN_EXPORT obj_res lean_io_prim_handle_get_line(b_obj_arg h, obj_arg
             }
             result.append(out);
         } else if (std::feof(fp)) {
+            ungetc(' ',fp); // reset the eof state
+            getc(fp); // remove the character introduced above
```

Comment:
Why not `clearerr`?

Reply:
simple answer: because I didn't know that one existed!